SSLDriverTests failures on Java 16

**Build scan**:

https://gradle-enterprise.elastic.co/s/nfsh3tlezlwba

**Repro line**:

```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch" -Dtests.seed=E6AA04E3D814FEAE -Dtests.security.manager=true -Dtests.locale=ca -Dtests.timezone=America/St_Barthelemy -Druntime.java=16
```

**Reproduces locally?**:

yes

**Applicable branches**:

I expect all branches running on Java 16

**Failure history**:

https://build-stats.elastic.co/goto/30a7ec831d753f0b753e6c60af3fa5ff

Started failing on January 8th

**Failure excerpt**:

```
   2> java.lang.AssertionError: Unexpected exception message: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
        at __randomizedtesting.SeedInfo.seed([E6AA04E3D814FEAE:7A3F56720CE43F06]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch(SSLDriverTests.java:175)

```

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java
@@ -29,6 +29,8 @@ import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntFunction;
 
+import static org.hamcrest.Matchers.is;
+
 public class SSLDriverTests extends ESTestCase {
 
     private final IntFunction<Page> pageAllocator = (n) -> new Page(ByteBuffer.allocate(n), () -> {});
@@ -159,20 +161,32 @@ public class SSLDriverTests extends ESTestCase {
         SSLEngine clientEngine = sslContext.createSSLEngine();
         SSLEngine serverEngine = sslContext.createSSLEngine();
 
-        String[] serverProtocols = {"TLSv1.2"};
+        final String[] serverProtocols;
+        final String[] clientProtocols;
+        final String expectedMessage;
+        if (inFipsJvm()) {
+            // fips JSSE does not support TLSv1.3 yet
+            serverProtocols = new String[]{"TLSv1.2"};
+            clientProtocols = new String[]{"TLSv1.1"};
+            expectedMessage = "org.bouncycastle.tls.TlsFatalAlert: protocol_version(70)";
+        } else if (JavaVersion.current().compareTo(JavaVersion.parse("16")) >= 0) {
+            // JDK16 https://jdk.java.net/16/release-notes does not permit protocol TLSv1.1 OOB
+            serverProtocols = new String[]{"TLSv1.3"};
+            clientProtocols = new String[]{"TLSv1.2"};
+            expectedMessage = "The client supported protocol versions [TLSv1.2] are not accepted by server preferences [TLS13]";
+        } else {
+            serverProtocols = new String[]{"TLSv1.2"};
+            clientProtocols = new String[]{"TLSv1.1"};
+            expectedMessage = "The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS12]";
+        }
+
         serverEngine.setEnabledProtocols(serverProtocols);
-        String[] clientProtocols = {"TLSv1.1"};
         clientEngine.setEnabledProtocols(clientProtocols);
         SSLDriver clientDriver = getDriver(clientEngine, true);
         SSLDriver serverDriver = getDriver(serverEngine, false);
 
         SSLException sslException = expectThrows(SSLException.class, () -> handshake(clientDriver, serverDriver));
-        String oldExpected = "Client requested protocol TLSv1.1 not enabled or not supported";
-        String jdk11Expected = "The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS12]";
-        String bctlsExpected = "org.bouncycastle.tls.TlsFatalAlert: protocol_version(70)";
-        boolean expectedMessage = oldExpected.equals(sslException.getMessage()) || jdk11Expected.equals(sslException.getMessage())
-            || bctlsExpected.equals(sslException.getMessage());
-        assertTrue("Unexpected exception message: " + sslException.getMessage(), expectedMessage);
+        assertThat(sslException.getMessage(), is(expectedMessage));
 
         // Prior to JDK11 we still need to send a close alert
         if (serverDriver.isClosed() == false) {

